src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -296,14 +472,47 @@ public void SetOptions(DacDeployOptions options)
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
                 var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-
+                // Note that we set excluded object types here since dacfx has this value as null;
                 if (prop != null)
                 {
-                    deployOptionsProp.SetValue(this, prop.GetValue(options));
+                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
                 }
             }
         }
 
+        /// <summary>
+        /// Sets the Value and Description to all properties
+        /// </summary>
+        /// <param name="prop"></param>
+        /// <param name="val"></param>
+        /// <param name="deployOptionsProp"></param>
+        public void SetGenericDeployOptionProps(PropertyInfo prop, DacDeployOptions options, PropertyInfo deployOptionsProp)
+        {
+            var val = prop.GetValue(options);
+            var attribute = prop.GetCustomAttributes<DescriptionAttribute>(true).FirstOrDefault();
+
+            if (prop != null && prop.PropertyType == typeof(ObjectType[]))
```

Comment:
You access prop above so the null check isn't necessary here (and you could shortcut it anyways with `prop?.PropertyType == typeof(ObjectType[])`)